buildspec/linuxTests.yml

```
@@ -41,7 +41,7 @@ phases:
                     # Ensure that "foo | run_and_report" fails correctly.
                     set -o pipefail
                     . buildspec/shared/common.sh
-                    2>&1 xvfb-run npm test --silent | run_and_report 2 \
+                    2>&1 xvfb-run npm test --silent; npm run mergeReports -- "$?" | run_and_report 2 \
                         'rejected promise not handled' \
```

Based on a local test of this command:

    2>&1 echo aaa; echo bbb | grep x

I suspect the above `;` may not correctly pipe all the output from `npm test` to `run_and_report` as intended.

To confirm that the piping to `run_and_report` is still working correctly, can you try temporarily changing this search patter to something that always outputs, like:
```suggestion
                        'deleteTestTempDirs' \
```

then CI should fail.

If it doesn't, we can fix the issue by changing the pipeline to something like (note: whitespace near the braces is required):

     { 2>&1 xvfb-run npm test --silent; npm run mergeReports -- "$?"; } | run_and_report 2 \

---

Just tested and that was indeed the case. I've switched to your suggestion and it seems to work as expected